src/ansys/units/systems.py

```diff
@@ -49,13 +55,13 @@ def __init__(self, name: str = None, base_units: list = None, unit_sys: str = No
             if not isinstance(unit, ansunits.Unit):
```

Comment:
So, this is the code where that strange function is used.  I think this validation check can be skipped for built-in/default unit systems.  The validation seems to:
1. Check if each of the units passed to the initializer are not an instance of a `Unit`.  If not, it constructs one, which is fine.
2. Check if the name of the unit is not "in" the configured default `_base_units`.   This seems fine.
3. Check if the UnitSystem object has an attribute with the same name as the passed unit already.  This is the check that is using that weird function we are discussing.  How is that possible for this condition to be true?  If we only expect a complete specification with all the units provided by the `base_units` argument then this check should be unnecessary.

I think you can remove the third check and that function we are discussing.

Reply:
I agree that the `base_dimension` function can be removed. I had the `UNIT_TYPE` check in just in case a list had a duplicate value. (e.g. ["kg", "m", "s", "K", "K", "radian", "mol", "cd", "A", "sr"]).

Reply:
Ah, ok... so here is where a unit test that checked this scenario might be useful (maybe I missed it).  I don't think there could be a duplicate like that because then the length would be incorrect and it would throw at the beginning of the initializer based on getting an incorrect length.  Now, the list might have a missing dimension, with one of them being duplicate:
eg: `["kg", "m", "s", "K", "K", "mol", "cd", "A", "sr"]`